uaclient/cli.py

```diff
@@ -905,6 +905,10 @@ def _detach(cfg: config.UAConfig, assume_yes: bool) -> int:
         ent = ent_cls(cfg=cfg, assume_yes=assume_yes)
         if ent.can_disable(silent=True):
             to_disable.append(ent)
+
+    # We will nake sure that services without dependencies are disabled first
+    to_disable.sort(key=lambda ent: len(ent.dependent_services))
```

Comment:
It feels like this will break in the scenario if we introduce any more complex dependency chains. We should discuss if we care, but it might be worth explicit ordering based on inspecting the dependent_service names so we don't break future us if we add more coplexity

- A depends on B
- B depends on C&D
- C depends on nothing
In this example, our sorting would try to disable C first, then A (which would prompt for D) and then B because it has a large dependend_services count.

Reply:
Yep, that would break. I thought about implementing a full topological sorting to address this issue, but I thought that it would be an overkill for now. But maybe you are right, maybe we should prepared for this approach.

Let me know if you think we should do that and I can implement the algorithm here, no problem at all

Reply:
Another reasonable approach is to ignore the dependent services altogether during detach, since we really don't care about their order anyway, we only care about not prompting the user here

Reply:
I think what you have makes sense at the moment we can address complexity if we need to in the future. While the 2nd suggestion you posted is compelling to just ignore disable prompts during detach, I still don't think that gets us out of the responsibility of ordering what we disable to prefer services without dependencies first.